docs/01-tooling/03-fast-and-isolated-JS-unit-tests.md

```diff
@@ -2,10 +2,159 @@
 
 ### Intro
 
+For testing frontend code, MediaWiki provides a browser based QUnit setup. For
+running the tests, you have to spin up MediaWiki, and load
+Special:JavaScriptTest in your browser, which will run all the QUnit tests for
+all the extensions and MediaWiki itself.
+
+From then on, you can filter by module or string, hide passed tests, and a few
+other things like select to load the assets in production mode or development
+mode (`debug=true`).
+
```

Comment:
Where's my patent-pending Joaquin screencast gif here showing how long it takes!?

Reply:
Added a TODO to record a gif or video for the section

Reply:
Makes sense, thanks!